### PR TITLE
Add deploy and REST API support to NeMo 2.0

### PR DESCRIPTION
**Pull request by athitten**
# What does this PR do ?

1. Adds `deploy` method from NeMo 1.0 as a `task` to NeMo 2.0 
2. This PR aims at minimum viable product to deploy and expose REST API with NeMo 2.0 and hence only adds `TRTLLM` backend route of deploy module to NeMo 2.0. Also, it only supports the basic arguments  for the export method. More optimizations like `paged_kv_cache`, `parallel_embedding`, `lora_plugin`, `multi_block_mode` etc., and other backends (like `in-framework`) will be supported in a subsequent PR.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by oyilmaz-nvidia — COMMENTED** [review]
Changes LGTM but the deploy module was not included in the pip installer. So, having references in the collections might break the code for those who uses the pip installer. Can you please check with @ericharper  to if this is Ok?

**Comment by ericharper** [comment]
> Changes LGTM but the deploy module was not included in the pip installer. So, having references in the collections might break the code for those who uses the pip installer. Can you please check with @ericharper to if this is Ok?

It's fine but needs to be import guarded

**Comment by athitten** [comment]
@oyilmaz-nvidia I have added import guard to avoid the ImportError with import deploy for users that use pip installation. Can you please review it, thanks!

File: nemo/collections/llm/api.py [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 from pathlib import Path
 from typing import Any, Callable, Optional, Union
 
 import pytorch_lightning as pl
+import uvicorn
 from typing_extensions import Annotated
 
 from nemo.collections.llm.utils import Config, task
+from nemo.deploy import DeployPyTriton
+from nemo.export.tensorrt_llm import TensorRTLLM
```
Comment (oyilmaz-nvidia) [inline nemo/collections/llm/api.py:13]: We might need a import guard here as well

Comment (athitten) [reply nemo/collections/llm/api.py]: @oyilmaz-nvidia added it in the latest commit. Let me know if it looks good.

**Review by oyilmaz-nvidia — APPROVED** [review]
LGTM

**Review by oyilmaz-nvidia — APPROVED** [review]

**Review by oyilmaz-nvidia — APPROVED** [review]

**Pull request merged**